automatic/python3/update.ps1

```diff
@@ -12,52 +14,146 @@ function global:au_SearchReplace {
     }
 
     ".\legal\VERIFICATION.txt" = @{
-      "(?i)(\s+x32:).*"     = "`${1} $($Latest.URL32)"
-      "(?i)(\s+x64:).*"     = "`${1} $($Latest.URL64)"
-      "(?i)(checksum32:).*" = "`${1} $($Latest.Checksum32)"
-      "(?i)(checksum64:).*" = "`${1} $($Latest.Checksum64)"
+      "(?i)(^\s*location on\:?\s*)\<.*\>"      = "`${1}<$release_files_url>"
+      "(?i)(\s+x32:).*"                        = "`${1} $($Latest.URL32)"
+      "(?i)(\s+x64:).*"                        = "`${1} $($Latest.URL64)"
+      "(?i)(checksum32:).*"                    = "`${1} $($Latest.Checksum32)"
+      "(?i)(checksum64:).*"                    = "`${1} $($Latest.Checksum64)"
+      "(?i)3.\d+(\s*Documentation archive\s*)\<.*\>" = "$($Latest.VersionTwoPart)`${1}<$($Latest.ZipUrl)>"
+      "(?i)(\s*can also be found at\s*)\<.*\>" = "`${1}<$($Latest.LicenseUrl)>"
     }
   }
 }
 
-function global:au_BeforeUpdate { rm tools\*.msi, tools\*.exe -ea 0; Get-RemoteFiles -Purge -NoSuffix }
+function global:au_BeforeUpdate {
+  rm tools\*.msi, tools\*.exe -ea 0
 
-function GetStreams() {
-  param($releaseUrls)
-  $streams = @{ }
+  Get-RemoteFiles -Purge -NoSuffix
 
-  $releaseUrls | % {
-    $version = $_.InnerText.Trim() -split ' ' | select -Index 1
-    #if ($version -match '[a-z]') { Write-Host "Skipping prerelease: '$version'"; return }
-    $versionTwoPart = $version -replace '([\d]+\.[\d]+).*', "`$1"
+  # download Python documentation archive
+  $webrequest = [System.Net.HttpWebRequest]::Create($Latest.ZipUrl)
+  $response_stream = $webrequest.GetResponse().GetResponseStream()
+  $zip = [IO.Compression.ZipArchive]::new($response_stream)
 
-    if ($streams.$versionTwoPart) { return }
+  # license text for legal/LICENSE.txt
+  $license_entry = $zip.GetEntry("$($Latest.ZipName)/license.txt")
+  $Latest.License = [System.IO.StreamReader]::new($license_entry.Open()).ReadToEnd()
```

Comment:
Since you are downloading the license and replacing the existing one, we would want to validate the license to ensure it hasn't been changed.

Reply:
It does change a few times a year (based on commit history: 6 in 2022, 6 in 2021, 3 in 2020, 5 in 2019, see: https://github.com/python/cpython/commits/main/Doc/license.rst) so this might be an additional maintenance burden when it is well-established that Python will remain under open source license, including software that's been incorporated into Python.